packages/react/src/components/AccountSettings/ChangePassword/ChangePassword.tsx

```diff
@@ -1,53 +1,165 @@
 } from './defaultComponents';
 import { ChangePasswordProps } from './types';
-import { FormValues } from '../types';
 
 const logger = new Logger('ChangePassword');
 
+const getIsDisabled = (
+  formValues: FormValues,
+  validationError: ValidationError
+): boolean => {
+  const { currentPassword, newPassword, confirmPassword } = formValues;
+
+  const hasEmptyField = !currentPassword || !newPassword || !confirmPassword;
+
+  const arePasswordsInvalid =
+    validationError.newPassword?.length > 0 ||
+    validationError.confirmPassword?.length > 0;
+
+  return hasEmptyField || arePasswordsInvalid;
+};
+
 function ChangePassword({
   onSuccess,
   onError,
+  validators,
 }: ChangePasswordProps): JSX.Element | null {
   const [errorMessage, setErrorMessage] = React.useState<string>(null);
   const [formValues, setFormValues] = React.useState<FormValues>({});
-
+  const [blurredFields, setBlurredFields] = React.useState<BlurredFields>([]);
+  const [validationError, setValidationError] = React.useState<ValidationError>(
+    {}
+  );
   const { user, isLoading } = useAuth();
 
-  /** Return null if Auth.getCurrentAuthenticatedUser is still in progress  */
-  if (isLoading) {
-    return null;
-  }
+  const isDisabled = getIsDisabled(formValues, validationError);
 
-  /** Return null if user isn't authenticated in the first place */
-  if (!user) {
-    logger.warn('<ChangePassword /> requires user to be authenticated.');
-    return null;
-  }
+  const passwordValidators: ValidatorSpec[] = React.useMemo(() => {
+    return validators ?? getDefaultPasswordValidators();
+  }, [validators]);
+
+  /*
+   * Note that formValues and other states are passed in as props so that
+   * it does not depend on whether or not those states have been updated yet
+   */
+  const validateNewPassword = React.useCallback(
+    (
+      formValues: FormValues,
+      blurredFields: BlurredFields,
+      eventType: InputEventType
+    ): string[] => {
+      const { newPassword } = formValues;
+
+      return runFieldValidators({
+        value: newPassword,
+        validators: passwordValidators,
+        eventType,
+        hasBlurred: blurredFields.includes('newPassword'),
+      });
+    },
+    [passwordValidators]
+  );
+
+  const validateConfirmPassword = React.useCallback(
+    (
+      formValues: FormValues,
+      blurredFields: BlurredFields,
+      eventType: InputEventType
+    ): string[] => {
+      const { newPassword, confirmPassword } = formValues;
+
+      const confirmPasswordValidators =
+        getDefaultConfirmPasswordValidators(newPassword);
+
+      return runFieldValidators({
+        value: confirmPassword,
+        validators: confirmPasswordValidators,
+        eventType,
+        hasBlurred: blurredFields.includes('confirmPassword'),
+      });
+    },
+    []
+  );
+
+  const runValidation = React.useCallback(
+    (
+      formValues: FormValues,
+      blurredFields: BlurredFields,
+      eventType: InputEventType
```

Comment:
This is me getting in to the weeds a bit but could just replace these params with an object and shared interface for use between `runValidation`, `validateNewPassword` and `validateConfirmPassword`:
```
const runValidation = React.useCallback((params: ValidateParams) => {
  const passwordErrors = validateNewPassword(params);
  const confirmPasswordErrors = validateConfirmPassword(params);

Reply:
👍, done in [6a7349b](https://github.com/aws-amplify/amplify-ui/pull/2900/commits/6a7349b94bae719dff691bdc27c7fa937888d72a)